src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -156,6 +156,10 @@ private static SingleNamespaceOrTypeDeclaration CreateImplicitClass(ImmutableSeg
 
         private static SingleNamespaceOrTypeDeclaration CreateSimpleProgram(GlobalStatementSyntax firstGlobalStatement, bool hasAwaitExpressions, bool isIterator, bool hasReturnWithExpression, ImmutableArray<Diagnostic> diagnostics)
         {
+            // PROTOTYPE: workaround for https://github.com/dotnet/roslyn/issues/67050 should be undone
+            var nameToken = firstGlobalStatement.GetFirstToken();
+            var nameLocation = new SourceLocation(nameToken.Kind() == SyntaxKind.None ? firstGlobalStatement : nameToken);
```

Comment:
>nameToken.Kind() == SyntaxKind.None ? firstGlobalStatement : nameToken [](http://example.com/codeflow?start=50&length=70)

What prevents us from always using location that starts at the beginning of the statement with length 1?

Reply:
My goal here is not to fix the issue, but to unblock myself with minimal change. This code will be undone per PROTOTYPE comment above.